daml-lf/api-type-signature/src/main/scala/lf/typesig/PackageSignature.scala

```diff
@@ -233,9 +233,9 @@ object PackageSignature {
     }
 
     (0 until newSignatures.size).foldLeft((s, newSignatures)) {
-      case (st @ (_, newSignatures), ifcK) =>
+      case (st @ (_, newSignatures1), ifcK) =>
         val ((s2, newSignatures2), newAtIfcK) =
-          newSignatures(ifcK).resolveRetroImplements(st)(findTpl)
+          newSignatures1(ifcK).resolveRetroImplements(st)(findTpl)
```

Comment:
naive question. Why ?

Reply:
Due to name shadowing the `newSignatures` on line 235 was not the same `newSignatures` as on line 236. Hopefully this makes that more obvious.